tests/integration_tests/tests/agentless_tests/test_executions.py

```diff
@@ -603,6 +604,31 @@ def test_cancel_execution_and_then_force_cancel(self):
 
         self._assert_execution_cancelled(execution, deployment_id)
 
+    def test_execute_and_kill_execution(self):
+        """Tests the kill execution option.
+
+        The function executes a new execution, writes the process to a local
+        file, kills the execution and asserts the process was also killed.
+        """
+        execution, node_inst_id, deployment_id = \
+            self._execute_from_resource(workflow_id='read_write',
+                                        workflow_params=None,
+                                        resource_file='dsl'
+                                                      '/read_write_workflows'
+                                                      '.yaml')
+        self._check_state('active_and_wrote', "Execution was expected to go "
+                                              "into 'active_and_wrote' "
+                                              "status", node_inst_id)
+        self._check_state('asleep', "Execution was expected to go"
```

Comment:
Modify _check_state to use `assertEquals` instead of sending the error message

Reply:
I tried to do it, but I couldn't find a way to retry the assertion process. So I thought of a work-around that still sends a Runtime error but in a basic format (kind of what David did). Tell me what you think. 

    def _check_state(self, expected_instance_state, node_inst_id):
        instance = self.client.node_instances.get(node_inst_id)
        for retry in range(30):
            if self.assertEquals(instance.state, expected_instance_state):
                break
            time.sleep(1)
        else:
            raise RuntimeError('Expected instance state is: {},'
                               'but the actual state is: {}'.format(
                                expected_instance_state, instance.state))